File: Sort/H011 Olympic Game/solution.py
class Solution(object):
    def canAttendMeetings(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: bool
        """
        medal = []
        for i in range(n):
            country, gold, silver, bronze = input()
            temp = [country, int]
            medal.append(temp)
            
        medal.sort(key = lambda x: (-x[1], -x[2], -x[3], x[0]))
        
        res = []
        for item in medal:
            res.append(item[0])
                        
        return res


if __name__ == '__main__':
    intervals = [[0,30],[5,10],[15,20]]
    assert(Solution().canAttendMeetings(intervals) == False)

    intervals = [[7,10],[2,4]]
    assert(Solution().canAttendMeetings(intervals) == True)

File: Sort/H011 Olympic Game/solution.cpp
#include <vector>
#include <algorithm>

struct CountryInfo{
	string countryName;
	int gold;
	int silver;
	int bronze;
};

bool Compare(CountryInfo first, CountryInfo second)
{
	if (first.gold != second.gold) return first.gold > second.gold;
	if (first.silver != second.silver) return first.silver > second.silver;
	if (first.bronze != second.bronze) return first.bronze > second.bronze;
	if (first.countryName != second.countryName) return first.countryName < second.countryName;
	return false;
}

int main()
{
	int n;
	cin >> n;
	CountryInfo temp;
	vector<CountryInfo> countries;
	
	for (int i = 0; i < n; i++) {
		countries.push_back(temp);
	}
	
	sort(countries.begin(), countries.end(), Compare);
	for (auto i: countries) {
		cout << i.countryName << endl;
	}
	
	return 0;
}